Validation/MtdValidation/plugins/BtlDigiHitsValidation.cc

```diff
@@ -0,0 +1,279 @@
+//
+/**\class BtlDigiHitsValidation BtlDigiHitsValidation.cc Validation/MtdValidation/plugins/BtlDigiHitsValidation.cc
+
+ Description: BTL DIGI hits validation
+
+ Implementation:
+     [Notes on implementation]
+*/
+
+#include <string>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+
+#include "DataFormats/FTLDigi/interface/FTLDigiCollections.h"
+
+#include "DataFormats/ForwardDetId/interface/BTLDetId.h"
+
+#include "Geometry/Records/interface/MTDDigiGeometryRecord.h"
+#include "Geometry/Records/interface/MTDTopologyRcd.h"
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeometry.h"
+#include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+
+#include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
+#include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
+
+
+class BtlDigiHitsValidation : public DQMEDAnalyzer {
+
+public:
+  explicit BtlDigiHitsValidation(const edm::ParameterSet&);
+  ~BtlDigiHitsValidation() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+private:
+  void bookHistograms(DQMStore::IBooker &,
+		      edm::Run const&,
+		      edm::EventSetup const&) override;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+  // ------------ member data ------------
+
+  const MTDGeometry* geom_;
+  const MTDTopology* topo_;
+
+  const std::string folder_;
+  const std::string infoLabel_;
+  const std::string btlDigisCollection_;
+
+  int eventCount_;
+
+  edm::EDGetTokenT<BTLDigiCollection> btlDigiHitsToken_;
+
+  // --- histograms declaration
+
+  MonitorElement* meNhits_[2];
+
+  MonitorElement* meHitCharge_[2];
+  MonitorElement* meHitTime_[2];
+
+  MonitorElement* meOccupancy_[2];
+
+  MonitorElement* meHitX_[2];
+  MonitorElement* meHitY_[2];
+  MonitorElement* meHitZ_[2];
+  MonitorElement* meHitPhi_[2];
+  MonitorElement* meHitEta_[2];
+
+  MonitorElement* meHitTvsQ_[2];
+  MonitorElement* meHitQvsPhi_[2];
+  MonitorElement* meHitQvsEta_[2];
+  MonitorElement* meHitQvsZ_[2];
+  MonitorElement* meHitTvsPhi_[2];
+  MonitorElement* meHitTvsEta_[2];
+  MonitorElement* meHitTvsZ_[2];
+
+};
+
+
+// ------------ constructor and destructor --------------
+BtlDigiHitsValidation::BtlDigiHitsValidation(const edm::ParameterSet& iConfig):
+  geom_(nullptr),
+  topo_(nullptr),
+  folder_(iConfig.getParameter<std::string>("folder")),
+  infoLabel_(iConfig.getParameter<std::string>("moduleLabel")),
+  btlDigisCollection_(iConfig.getParameter<std::string>("btlDigiHitsCollection")),
+  eventCount_(0) {
+
+  btlDigiHitsToken_ = consumes <BTLDigiCollection> (edm::InputTag(std::string(infoLabel_),
+								  std::string(btlDigisCollection_)));
```

Comment:
Could `edm::InputTag` be used directly in the configuration instead of two strings?

Reply:
Done.